### PR TITLE
Issue-51: ut coverage controller and upgrade

### LINKED ISSUE #51: Add Ut coverage for bookkeepercluster_controller.go and upgrade.go files
### Description
We don't have any unit test coverage for bookkeepercluster_controller.go and upgrade.go files

### Importance
Nice to have.

### Location
Test files should be added in the pkg folder

### Suggestions for an improvement
Add  bookkeepercluster_controller_test.go and upgrade_test.go files in the pkg/controller folder

### PR DESCRIPTION
**Pull request by Prabhaker24**
### Change log description
This pr adds ut coverage for  controller.go and upgrade.go files 

### Purpose of the change
This fixes #51 

### What the code does
It adds ut coverage for controller.go and upgrade.go files 

### How to verify it
Ru make test command and all unit test should run successfully

### CODE REVIEW

File: pkg/controller/bookkeepercluster/upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+					boolean bool
+					pod     []*corev1.Pod
+				)
+				BeforeEach(func() {
+					testpod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "test"}, Spec: v1.PodSpec{Containers: []v1.Container{{Image: "testimage"}}},
+						Status: v1.PodStatus{
+							ContainerStatuses: []v1.ContainerStatus{
+								{
+									Name: "test",
+									State: v1.ContainerState{
```
Comment (anishakj) [inline pkg/controller/bookkeepercluster/upgrade_test.go:266]: you can check for pod non-faulty case as well here

File: pkg/controller/bookkeepercluster/bookkeepercluster_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+					b.WithDefaults()
+					//to ensure the client get for BookKeepercluster fails
+					err = r.syncBookieSize(b)
+					_, _ = r.Reconcile(req)
+					b.Spec.Replicas = 5
+					client.Update(context.TODO(), b)
+					_, err1 = r.Reconcile(req)
+				})
+
+				It("should give error", func() {
```
Comment (anishakj) [inline pkg/controller/bookkeepercluster/bookkeepercluster_controller_test.go:138]: can we check for specific error

File: pkg/controller/bookkeepercluster/upgrade_test.go (hunk 1 above)
Comment (Prabhaker24) [reply pkg/controller/bookkeepercluster/upgrade_test.go:265]: Done.

File: pkg/controller/bookkeepercluster/bookkeepercluster_controller_test.go (hunk 2 above)
Comment (Prabhaker24) [reply pkg/controller/bookkeepercluster/bookkeepercluster_controller_test.go:139]: Done.

File: pkg/controller/bookkeepercluster/bookie_test.go [hunk 3]
```diff
@@ -72,71 +71,54 @@ var _ = Describe("Bookie", func() {
...
-					_ = bookkeepercluster.MakeBookieConfigMap(bk)
-					Ω(err).Should(BeNil())
+					cm := bookkeepercluster.MakeBookieConfigMap(bk)
+					Ω(cm.Name).Should(Equal(util.ConfigMapNameForBookie(bk.Name)))
 				})
 
 				It("should create a stateful set", func() {
-					_ = bookkeepercluster.MakeBookieStatefulSet(bk)
-					Ω(err).Should(BeNil())
+					ss := bookkeepercluster.MakeBookieStatefulSet(bk)
```
Comment (anishakj) [inline pkg/controller/bookkeepercluster/bookie_test.go:119]: Check for journal and ledger directories as well here

File: pkg/controller/bookkeepercluster/upgrade_test.go [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+								},
+							},
+						},
+					}
+					r.client.Create(context.TODO(), testpod)
+					r.client.Get(context.TODO(), types.NamespacedName{Name: "test", Namespace: "default"}, testpod)
+					pod = append(pod, testpod)
+					boolean, err = r.checkUpdatedPods(pod, "0.7.1")
+				})
+				It("Error should not be nil and bool value should be false", func() {
```
Comment (anishakj) [inline pkg/controller/bookkeepercluster/upgrade_test.go:303]: change to error should be nil

**Pull request merged**